packages/terra-form-select/src/combobox/Menu.jsx

```diff
@@ -356,14 +356,16 @@ class Menu extends React.Component {
     }
 
     if (element) {
+      const index = active.getAttribute('index');
+      const totalOptions = optGroupElement ? optGroupElement.props.children.length : document.querySelectorAll('[data-terra-select-option]').length;
```

Comment:
I don't think we should query the entire document. This might work due to the conditional render of the dropdown only if open, but it feels really dangerous. We should find a safer way to get the child count of this specific menu

Reply:
When we don't have any `optGroup` or if we have `multiple optGroups`, then these changes work fine.

But when we have combinations of `non-optGroups and optGroups`, I think there is an issue. Attaching a gif to display the problem.

![ezgif com-gif-maker](https://user-images.githubusercontent.com/29273628/110355401-2ff89200-805f-11eb-9e64-277c87bf2922.gif)
Here the `totalOptions` jumps from 10 to 5, the moment an optGroup is encountered.

Proposed solution:

![ezgif com-gif-maker (1)](https://user-images.githubusercontent.com/29273628/110355554-60d8c700-805f-11eb-9bce-e42c4c83d968.gif)
We can change the sentencing, But I believe this will be a better approach.

Here's the snippet for this:
```jsx
if (element) {
  const index = active.getAttribute('index');
  const totalOptions = this.props.children[1].length;
  const optGroupElementsLength = optGroupElement ? optGroupElement.props.children.length : undefined;
  if (element.props.display === '' && element.props.value === '') {
    // Used for case where users selects clear option and opens
    // dropdown again and navigates to clear option
    visuallyHiddenComponent.current.innerText = clearSelectTxt;
  } else if (this.isActiveSelected()) {
    visuallyHiddenComponent.current.innerText = optGroupElementsLength ? `${intl.formatMessage({ id: 'Terra.form.select.selectedText' }, { text: displayText })}, option ${index} of ${optGroupElementsLength} group options, of total ${totalOptions} options` : `${intl.formatMessage({ id: 'Terra.form.select.selectedText' }, { text: displayText })} ${index} of total ${totalOptions} options`;
  } else {
    visuallyHiddenComponent.current.innerText = optGroupElementsLength ? `${`${displayText}, option ${index}`} of ${optGroupElementsLength} group options, of total ${totalOptions} options` : `${`${displayText}, option ${index}`} of total ${totalOptions} options`;
  }
}
```

This will also resolve the issue of using the `document` mentioned above.

Reply:
+1 to the intent of the proposals above and the problems found.

After much testing and researching, we'll want to change the options within option groups from being announcing with their own index count and total, but flatten the list to sound like it is only one list overall, with a single total and unique index for each option item, similar as to how we are already flattening the OptGroup label as part of the `displayText`.

For your example listed above with the combination of options and option-groups, the displayText plus setsize information would read back as:
```
Blue (1 of 10),
Cyan (2 of 10),
Azul (4 of 10),
Aqua (5 of 10),
Group Shades of Green, Green (6 of 10),
Group Shades of Green, Forest Green (7 of 10),
Group Shades of Green, Neon Green (9 of 10),
Group Shades of Green, Emerald 10 of 10),
```

_Side note: We may need to revisit disabled option items, since they are currently never read back in either the menu or frame and the count ._

If we needed a static total count (e.g. for the legacy single-select) the value could be passed in to the menu as a prop from the frame, which already has it defined.

But since the combobox, multi, and tag can all be filtered based on what is typed in the input, the totalOptions, the setsize of the list changes dynamically, so it could make the most sense to just set both the index/posinset and the total/setsize as attributes on each option, such that when the list is filtered, the attribute values can be updated, something more like:

```
    if (element) {
      const active = this.menu.querySelector('[data-select-active]');
      const posinset = active.getAttribute('aria-posinset');
      const setsize = active.getAttribute('aria-setsize');
      if (element.props.display === '' && element.props.value === '') {
        // Used for case where users selects clear option and opens
        // dropdown again and navigates to clear option
        visuallyHiddenComponent.current.innerText = clearSelectTxt;
      } else if (this.isActiveSelected()) {
        visuallyHiddenComponent.current.innerText = `${intl.formatMessage({ id: 'Terra.form.select.selectedText' }, { text: displayText })} ${intl.formatMessage({ id: 'Terra.form.select.displayPositionInSetSize' }, { indexNumber: posinset, totalNumber: setsize })},`;
      } else {
        visuallyHiddenComponent.current.innerText = `${displayText} ${intl.formatMessage({ id: 'Terra.form.select.displayPositionInSetSize' }, { indexNumber: posinset, totalNumber: setsize })},`;
      }
    }
```

So using the same list above, typing "Gr" in the input, then navigating down into the list would read:
```
Group Shades of Green, Green (1 of 4),
Group Shades of Green, Forest Green (2 of 4),
Group Shades of Green, Neon Green (4 of 4),
```

_Again, how to handle disabled option items may need to be revisited later._